src/main/scala/com/target/data_validator/ValidatorTable.scala

```diff
@@ -210,7 +211,18 @@ case class ValidatorHiveTable(
 
   override def getDF(session: SparkSession): Try[DataFrame] = {
     logger.info(s"Opening table: $db.$table")
-    Try(session.table(s"$db.$table"))
+    useHWC match {
+      case Some(x) if x =>
+        val hive = com.hortonworks.hwc.HiveWarehouseSession.session(session).build()
```

Comment:
the example api usage does it this way without the `import`, not sure why. See [README](https://github.com/hortonworks-spark/spark-llap#api-usage)

I don't have access to a Hadoop cluster that has hive3 setup. 
I'm wondering if you can use reflection to try and load the `com.hortonworks.hwc.HiveWarehouseSession.HiveWarehouseSession` class using the ClassLoader,  and if it fails, fall back to using  `session.table()` 
That would eliminate the need for the `useHWC` flag, and the `compileonly` `sbt`.

Reply:
The availability of `com.hortonworks.hwc.HiveWarehouseSession` at runtime depends on whether the path to the `hive-warehouse-connector` jar has been passed using the `--packages` option in the `spark-submit` command. Since this is user-driven, we cannot rely on it. In case the jar path is not passed by the user, I think it is better to fail with a CNFE (which will happen in the current implementation) instead of trying to use SparkSession as a fallback.

Reply:
I don't care for the `useHWC` flag myself, but I'd like to punt on potentially better solutions for now.